### ISSUE TITLE
Investigate the View Payments API (1 hour)

### ISSUE THREAD

**Issue by jason-gcio** [department-of-veterans-affairs/va.gov-team#9371]
## Goal 
We want to determine what is in this call to see how we can best move forward in developing "View your VA payment history."

## Considerations
- This is a single (BGS) service call that returns a list of things
- Includes a link that drills down to the payment details?
- Per SK: http://bepdev.vba.va.gov/vrm-ws/PaymentInformationService?WSDL
  - And our ESCP request for this has been approved

## Tasks
-Hit the endpoint, see what it returns and hash out some next steps with Jim

## Acceptance Criteria
- [ ] Results from these findings have been documented and shared with the team.

**Comment by digitaldrk** [comment]
I know this was only time-boxed to 1hr but I could not let it go until I retrieved all data 1 call or more. I _STILL_ could not get returned payment data back from any call, below is my summary.

## TL;DR
Inconclusive. I cannot get returned payment data back at all from any call using user Jerry Brooks who has returned payment data populating in eben.

## Call `PaymentInformationService` -> `retrievePaymentSummary`
User: Jerry Brooks

### Payments Received
* Date `(payment) -> paymentDate` 
* Amount `(payment) -> paymentAmount` 
* Type (ex. Compensation & Pension Recurring) `(payment) -> paymentType`
* Method (ex. Direct Deposit) `(payment) -> addressEFT -> accountType` ? Ex ‘Checking’
* Bank (ex. USAA FEDERALO SAVINGS BANK) `(payment) -> addressEFT -> bankName`
* Account (I think this is account number) `(payment) -> addressEFT -> accountNumber`

There is a `returnPayment` object inside the `payment` object but I cannot find payment payload with data inside this even when referencing eben site with user that has return payment info.

Returned empty values include:
* `checkTraceNumber`
* `returnReason`

## Call `PaymentInformationService `  ->  `retrieveReturnedPaymentSummaryFilter`
User: Jerry Brooks

This service seems logically to be the service to retrieved return payment data. However, I cannot get data to return even when using a date span of 1900 -> 2020

### Payments Returned
* Date
* Amount
* Type
* Method
* Bank Name
* Account

**Comment by jason-gcio** [comment]
![Payment_History_VA_DoD_eBenefits](https://user-images.githubusercontent.com/53826255/82735321-d94be280-9cee-11ea-95e8-db987b5d78b3.png)

**Comment by jason-gcio** [comment]
![Payment_History_VA_DoD_eBenefits-cropped](https://user-images.githubusercontent.com/53826255/82735360-35166b80-9cef-11ea-81bf-909ae215047d.png)

**Comment by jason-gcio** [comment]
Great, thanks Derek. Posted a screenshot of returns we see for Jerry in the PINT environment for EBN staging. 

Next question or step is to determine where returned payments come from.  We might need to ask some EVSS/BGS folks; we'll catch up with SK on it next week.
